src/python/WMCore/Services/DBS/DBSUtils.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python
+"""
+_DBSUtils_
+
+set of common utilities for DBS3Reader
+
+"""
+import json
+import urllib
+from urllib.parse import urlparse
+from urllib.parse import parse_qs
+from collections import defaultdict
+
+from Utils.CertTools import cert, ckey
+from dbs.apis.dbsClient import aggFileLumis, aggFileParents
+from WMCore.Services.pycurl_manager import getdata as multi_getdata
+
+
+def dbsListFileParents(dbsUrl, blocks, logger):
+    """
+    Concurrent counter part of DBS listFileParents API
+
+    :param dbsUrl: DBS URL
+    :param blocks: list of blocks
+    :param logger: logger
+    :return: list of file parents
+    """
+    urls = ['%s/fileparents?block_name=%s' % (dbsUrl, b.replace('#', '%23')) for b in blocks]
```

Comment:
Instead of replacing string, I'd suggest to stick to the standard way of encoding urls (based on urllib). In this case, you want to use `urllib.parse.quote_plus`.

Reply:
that's fine